packages/front-end/components/Experiment/EditDOMMutationsModal.tsx

```diff
@@ -54,26 +51,25 @@ const EditDOMMutatonsModal: FC<{
   );
 
   const setDOMMutation = useCallback(
-    (index: number, updates) => {
-      setNewVisualChange({
-        ...newVisualChange,
-        domMutations: newVisualChange.domMutations.map((m, i) =>
+    (index: number, updates: DOMMutation) => {
+      setNewVisualChange((prevVisualChange) => ({
+        ...prevVisualChange,
+        domMutations: prevVisualChange.domMutations.map((m, i) =>
           i === index ? updates : m
         ),
-      });
+      }));
     },
-    [newVisualChange, setNewVisualChange]
+    [setNewVisualChange]
   );
 
-  const setDOMMutationStr = useCallback(
-    (index: number, str: string) => {
-      setNewDOMMutationStr((strs) => {
-        const newStrs = [...strs];
-        newStrs[index] = str;
-        return newStrs;
+  const addDOMMutation = useCallback(
+    (updates: DOMMutation) => {
+      setNewVisualChange({
+        ...newVisualChange,
+        domMutations: [...newVisualChange.domMutations, updates],
       });
     },
-    [setNewDOMMutationStr]
+    [newVisualChange, setNewVisualChange]
```

Comment:
Nit: Switch `setNewVisualChange` to use the callback version like you did above.  Then, you can remove `newVisualChange` from the dependencies array.